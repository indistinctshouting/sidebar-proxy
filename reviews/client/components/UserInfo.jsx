import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUserFriends } from '@fortawesome/free-solid-svg-icons';
import { faStar } from '@fortawesome/free-solid-svg-icons';
import { faCamera } from '@fortawesome/free-solid-svg-icons';
import styled from 'styled-components';

const SidebarWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr 2fr;
  padding: 0 15px;
`;

const UserInfoWrapper = styled.div`
  display: grid;
  grid-template-rows: repeat(5, 18px);
  grid-row-gap: 2px;
`;

const UserInfoText = styled.span`
  font-size: 12px;
  color: #666;
  font-weight: 400;
`;

const LocationText = styled.span`
  font-size: 1vw;
  font-weight: 700;
  color: #333;
  white-space: nowrap;
`;

const UserInfoTextBold = styled(UserInfoText)`
  font-weight: 700;
`;

const NameText = styled(UserInfoText)`
  font-size: 14px;
  line-height: 1.28571em;
  font-weight: 700;
  color: #0073bb;
  :hover {
    text-decoration: underline;
  }
`;

const EliteText = styled(UserInfoText)`
  font-weight: 700;
  color: #f15c00;
`;

const Avatar = styled.img`
  width: 60px;
  height: 60px;
  margin-right: 9px;
  border-radius: 4px;
`;

const InfoWrapper = styled.div`
  display: grid;
  grid-template-columns: 18px 100%;
  grid-column-gap: 4px;
  .info-icon {
    width: 18px;
    height: 18px;
    margin-right: 4px;
    color: #f15c00;
  }
`;


const UserInfo = (props) => (
  <SidebarWrapper>
    <Avatar src={props.owner.picture} />
    
    <UserInfoWrapper>
      <div className="user-name">
        <NameText>
          {props.owner.name}
        </NameText>
      </div>
      <div className="user-location">
        <LocationText>
          {props.owner.location}
        </LocationText>
      </div>
      <InfoWrapper>
        <span className="info-icon">
          <FontAwesomeIcon icon={faUserFriends}/>
        </span>
        <UserInfoText>
          <UserInfoTextBold>{props.owner.friends}</UserInfoTextBold> Friends
        </UserInfoText>
      </InfoWrapper>
      <InfoWrapper>
        <span className="info-icon">
          <FontAwesomeIcon icon={faStar}/>
        </span>
        <UserInfoText>
          <UserInfoTextBold>{props.owner.reviewCount}</UserInfoTextBold> Reviews
        </UserInfoText>
      </InfoWrapper>
      <InfoWrapper>
        <span className="info-icon">
          <FontAwesomeIcon icon={faCamera} />
        </span>
        <UserInfoText>
          <UserInfoTextBold>{props.owner.photos}</UserInfoTextBold> Photos
        </UserInfoText>
      </InfoWrapper>
      <div>
        <EliteText>
          {props.owner.elite ? 'Elite \'18' : ''}
        </EliteText>
      </div>
    </UserInfoWrapper>
  </SidebarWrapper>
);

export default UserInfo;