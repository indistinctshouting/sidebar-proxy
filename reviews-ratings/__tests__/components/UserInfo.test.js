import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserInfo from '../../client/components/UserInfo';

const fakeOwner = { 
  picture: 'https://s3.amazonaws.com/uifaces/faces/twitter/dshster/128.jpg',
  name: 'Giovanny Reinger ',
  location: 'Caliton, VT',
  friends: 22,
  reviewCount: 112,
  photos: 37,
  checkIns: 8,
  elite: false 
};

describe('<UserInfo />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<UserInfo owner={fakeOwner}/>);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});