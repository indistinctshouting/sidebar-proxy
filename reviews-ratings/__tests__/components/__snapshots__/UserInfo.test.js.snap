// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo /> render() renders the component 1`] = `
<div
  className="review-sidebar"
>
  <img
    className="user-avatar"
    src="https://s3.amazonaws.com/uifaces/faces/twitter/dshster/128.jpg"
  />
  <div
    className="user-info"
  >
    <div
      className="user-name"
    >
      <span
        className="user-info-text name-text"
      >
        Giovanny Reinger 
      </span>
    </div>
    <div
      className="user-location"
    >
      <span
        className="user-info-text location-text"
      >
        Caliton, VT
      </span>
    </div>
    <div
      className="user-friends"
    >
      <span
        className="friends-icon"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                640,
                512,
                Array [],
                "f500",
                "M192 256c61.9 0 112-50.1 112-112S253.9 32 192 32 80 82.1 80 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C51.6 288 0 339.6 0 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zM480 256c53 0 96-43 96-96s-43-96-96-96-96 43-96 96 43 96 96 96zm48 32h-3.8c-13.9 4.8-28.6 8-44.2 8s-30.3-3.2-44.2-8H432c-20.4 0-39.2 5.9-55.7 15.4 24.4 26.3 39.7 61.2 39.7 99.8v38.4c0 2.2-.5 4.3-.6 6.4H592c26.5 0 48-21.5 48-48 0-61.9-50.1-112-112-112z",
              ],
              "iconName": "user-friends",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
      <span
        className="user-info-text"
      >
        <b>
          22
        </b>
         Friends
      </span>
    </div>
    <div
      className="user-reviews"
    >
      <span
        className="reviews-icon"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                576,
                512,
                Array [],
                "f005",
                "M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z",
              ],
              "iconName": "star",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
      <span
        className="user-info-text"
      >
        <b>
          112
        </b>
         Reviews
      </span>
    </div>
    <div
      className="user-photos"
    >
      <span
        className="photos-icon"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f030",
                "M512 144v288c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V144c0-26.5 21.5-48 48-48h88l12.3-32.9c7-18.7 24.9-31.1 44.9-31.1h125.5c20 0 37.9 12.4 44.9 31.1L376 96h88c26.5 0 48 21.5 48 48zM376 288c0-66.2-53.8-120-120-120s-120 53.8-120 120 53.8 120 120 120 120-53.8 120-120zm-32 0c0 48.5-39.5 88-88 88s-88-39.5-88-88 39.5-88 88-88 88 39.5 88 88z",
              ],
              "iconName": "camera",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </span>
      <span
        className="user-info-text"
      >
        <b>
          37
        </b>
         Photos
      </span>
    </div>
    <div>
      <span
        className="user-info-text elite"
      />
    </div>
  </div>
</div>
`;
